pyproject.toml

```diff
@@ -5,7 +5,7 @@ requires = [
   "setuptools_scm[toml]>=7.0",
   "wheel",
   "oldest-supported-numpy",
-  "Cython"
+  "Cython>=3.0"
```

Comment:
> @trexfeathers I think this may possibly be all that is needed ?
> 
> So far, this adopts exactly what Bill proposed :
> 
> * insist on Cython>=3
> * remove the "language_level" setting.
> 
> But .. I'm still not sure it wouldn't be better to simply _unpin_ Cython, and leave the language-level explicit (which is then not superfluous).
> 
> Particularly as I see that the build does raises a warning about the default having changed recently. `/Users/runner/work/cf-units/cf-units/.tox/py311-osx-test/lib/python3.11/site-packages/Cython/Compiler/Main.py:381: FutureWarning: Cython directive 'language_level' not set, using '3str' for now (Py3). This has changed from earlier releases! File: /Users/runner/work/cf-units/cf-units/cf_units/_udunits2.pxd`
> 
> What do you think ?

We're talking a lot about maintenance burden at the moment. Bugs in cf-units that are specific to `cython<3` could be a real maintenance headache. TLDR just because software _can_ support older versions doesn't mean that human developers _want_ to:

- They are increasingly likely as we continue to develop cf-units using the latest releases of Cython, while user transition may be more gradual.
- It would be difficult for us to replicate and diagnose a reported bug like this - unless we have gathered a particularly detailed bug report we will probably be doing our testing using the latest version of Cython.
- Once we were aware that the bug was not present with `cython>=3`, we would be reluctant to fix it anyway. Including the minimum pin saves everyone time, and possibly saves disagreements about 'false advertising'.

**I would like to include this minimum pin.**

Reply:
OK then that just means we leave it as it is.